runtime/src/snapshot_utils.rs

```diff
@@ -145,6 +145,8 @@ pub struct BankSnapshotInfo {
     pub snapshot_path: PathBuf,
     /// Type of the snapshot
     pub snapshot_type: BankSnapshotType,
+    /// Path to the status cache (transaction history)
+    pub status_cache_path: PathBuf,
```

Comment:
Yes, if there is the directory path is in SnapshotInfo, then we can derive the file and status cache from it.  For the snapshot file we also need to pass the pre or post info to derive the file path.  There will be many small changes at many places.  I am not sure if it is worth doing it here.

Reply:
An intermediate option is to add the directory path into it, and derive status_cache_path from there, without touching the snapshot file path.  Latter we could make another PR to derive the snapshot file path from the snapshot directory path.  Let me know if this is OK.  @brooksprumo

Reply:
Ok, add the dir into BankSnapshotInfo, removed status_cache_path.

Reply:
I cannot remove snapshot_path because it cannot be derived from only snapshot_dir.  the path includes "pre" or "post" which is determined elsewhere.  It is still better to derive status_cache_path from snapshot_dir.

Reply:
I feel Option #2 is better, because there is no duplicated info.  Trying to do it that way.  

That means adding a function to derive snapshot_path.  all the previous accessing code snapshot_info.snapshot_path becomes .snapshot_path().

I got this error.  Any idea how to get around it?

```
error[E0515]: cannot return reference to temporary value
    --> runtime/src/snapshot_utils.rs:1970:26
     |
1970 |             .map(|paths| paths.snapshot_path().display()),
     |                          ---------------------^^^^^^^^^^
     |                          |
     |                          returns a reference to data owned by the current function
     |                          temporary value created here

Reply:
Removed .display(), following Brooks' suggestion offline.  Now it is done as Option #2.

snapshot_path() function may be evaluated multiple times.  Not sure if we an optimized with some kind of static variables in the function.